independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java

```diff
@@ -706,6 +714,46 @@ static boolean isPrimitiveClassName(String className) {
         return PRIMITIVE_CLASS_NAMES.contains(className);
     }
 
+    static boolean isPrimitiveWrapperType(Type type) {
+        if (type.kind() == Kind.CLASS) {
+            return PRIMITIVE_WRAPPERS.contains(type.name());
+        }
+        return false;
+    }
+
+    /**
+     * Emits a bytecode instruction to load the zero value of given {@code primitive} type
```

Comment:
I think that both CDI and JLS specs talk about "default values" and not "zero values" but that's a pretty useless nitpicking from my side :D

Reply:
Correct. I wanted to rebase this PR anyway after #30544 is merged, so I'll fix this :-)

Reply:
Done.